pyro/generic/generic.py

```diff
@@ -36,55 +73,83 @@ def pyro_backend(*aliases, **new_backends):
     or by providing a dict mapping module name to backend module name.
     Standard backends include: pyro, minipyro, funsor, and numpy.
     """
+    global _BACKEND
```

Comment:
I'd rather avoid duplicate state if possible. We should not be if-else branch on a name in this file, because that pattern is not extensible.

Reply:
+1, I think this is ugly. We should instead figure out what the behavior of the seed handler be in Pyro and remove this code.